.github/workflows/test.yml

```diff
@@ -0,0 +1,145 @@
+name: codechecker-tests
+
+# Triggers the workflow on push or pull request events.
+on: [push, pull_request]
+
+jobs:
+  lint:
+    name: Linters (pylint, pycodestyle)
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - name: Install dependencies
+        run: |
+          pip install $(grep -iE "pylint|pycodestyle" analyzer/requirements_py/dev/requirements.txt)
+      - name: Run tests
+        run: make pylint pycodestyle
+
+  analyzer:
+    name: Analyzer
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+
+      - name: Install dependencies
+        run: sh .github/workflows/install-deps.sh
+
+      - name: Build the package
+        run: |
+          make pip_dev_deps
+          BUILD_UI_DIST=NO make package
+
+      - name: Run analyzer tests
+        run: |
+          make -C analyzer \
+            test_unit \
+            test_functional \
+            test_tu_collector \
+            test_merge_clang_extdef_mappings \
+            test_statistics_collector \
+            test_build_logger
+
+  web:
+    name: Web
+    runs-on: ubuntu-latest
```

Comment:
Would it be better to select some specific ubuntu version? What is the latest used here? LTS latest (20.04)?
The install_deps.sh assumes that the distribution is ubuntu bionic.